src/coreclr/nativeaot/Runtime/unix/UnwindHelpers.cpp

```diff
@@ -1088,6 +1090,318 @@ void Registers_REGDISPLAY::setVectorRegister(int num, libunwind::v128 value)
 
 #endif // TARGET_LOONGARCH64
 
+#if defined(TARGET_RISCV64)
+
+// Shim that implements methods required by libunwind over REGDISPLAY
+struct Registers_REGDISPLAY : REGDISPLAY
+{
+    inline static int  getArch() { return libunwind::REGISTERS_RISCV; }
+    inline static int  lastDwarfRegNum() { return _LIBUNWIND_HIGHEST_DWARF_REGISTER_RISCV; }
+
+    bool        validRegister(int num) const;
+    bool        validFloatRegister(int num) { return false; };
+    bool        validVectorRegister(int num) const;
+
+    uint64_t    getRegister(int num) const;
+    void        setRegister(int num, uint64_t value, uint64_t location);
+
+    double      getFloatRegister(int num) const { abort(); }
+    void        setFloatRegister(int num, double value) { abort(); }
+
+    libunwind::v128    getVectorRegister(int num) const;
+    void        setVectorRegister(int num, libunwind::v128 value);
+
+    uint64_t    getSP() const         { return SP; }
+    void        setSP(uint64_t value, uint64_t location) { SP = value; }
+    uint64_t    getIP() const         { return IP; }
+    void        setIP(uint64_t value, uint64_t location) { IP = value; }
+    uint64_t    getFP() const         { return *pFP; }
+    void        setFP(uint64_t value, uint64_t location) { pFP = (PTR_uintptr_t)location; }
+};
+
+inline bool Registers_REGDISPLAY::validRegister(int num) const {
+    if (num == UNW_REG_SP || num == UNW_RISCV_X2)
+        return true;
+
+    if (num == UNW_REG_IP)
+        return true;
+
+    if (num >= UNW_RISCV_X0 && num <= UNW_RISCV_X31)
+        return true;
+
+    return false;
+}
+
+bool Registers_REGDISPLAY::validVectorRegister(int num) const
+{
+    // Vector registers are mapped to floating-point registers F24 to F31
+    if (num >= UNW_RISCV_F24 && num <= UNW_RISCV_F31)
+        return true;
+
+    return false;
```

Comment:
```suggestion
    // Vector registers currently unsupported
    return false;
```
It's a separate register bank v0-v31 on RISC-V, we can worry about it when we bring up the "V" extension.